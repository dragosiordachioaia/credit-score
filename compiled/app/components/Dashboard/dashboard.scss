body {
  background: no-repeat top;
  background-size: cover;
  // background-image: url('/_background--dashboard.png');
  background-image: url('/_background--dashboard.png');
  // background-image: url('/bg_green.jpg');
  // background-image: url('/bg_forest.jpg');
  width: 100vw;
  height: 100vh;
  margin: 0;
  font-family: AktivGrotesk, Helvetica, 'sans-serif';
  overflow: hidden;
}

p {
  font-weight: 100;
  margin: 0;
}

b {
  font-weight: 300;
}

h1,
h2,
h3,
h4,
h5,
h6 {
  font-weight: 100;
  margin: 0;
}
