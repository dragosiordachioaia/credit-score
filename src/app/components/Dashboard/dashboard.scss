body {
  background: no-repeat top;
  background-size: cover;
  background-image: url('/_background--dashboard.png');
  // background-image: url('/background--dashboard.jpg');
  width: 100vw;
  height: 100vh;
  margin: 0;
}
