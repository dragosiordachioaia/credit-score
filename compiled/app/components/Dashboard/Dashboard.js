"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _bem = _interopRequireDefault(require("../../utils/bem"));

var _fetch = require("../../utils/fetch");

var _Constants = require("../../constants/Constants");

var _Slide = _interopRequireDefault(require("../Slide/Slide"));

require("./dashboard.scss");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } _setPrototypeOf(subClass.prototype, superClass && superClass.prototype); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.getPrototypeOf || function _getPrototypeOf(o) { return o.__proto__; }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

var cn = (0, _bem.default)({
  block: 'dashboard'
});

var Dashboard =
/*#__PURE__*/
function (_Component) {
  function Dashboard(props) {
    var _this;

    _classCallCheck(this, Dashboard);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Dashboard).call(this, props));
    _this.state = {
      score: null,
      error: null,
      windowWidth: 0
    };
    _this.fetchScoreData = _this.fetchScoreData.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    _this.displayMainSlide = _this.displayMainSlide.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    _this.displayMobileView = _this.displayMobileView.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    _this.displayNonMobileView = _this.displayNonMobileView.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    _this.onWindowResize = _this.onWindowResize.bind(_assertThisInitialized(_assertThisInitialized(_this)));
    return _this;
  }

  _createClass(Dashboard, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      if (!this.state.score && !this.state.error) {
        this.fetchScoreData();
      }

      if (window) {
        window.addEventListener("resize", this.onWindowResize);
      }
    }
  }, {
    key: "onWindowResize",
    value: function onWindowResize() {
      this.setState({
        windowWidth: window.innerWidth
      });
    }
  }, {
    key: "fetchScoreData",
    value: function fetchScoreData() {
      var _this2 = this;

      (0, _fetch.getJSON)(_Constants.SCORE_DATA_URL).then(function (response) {
        _this2.setState({
          score: response
        });
      }, function (error) {
        _this2.setState({
          error: error
        });
      }).catch(function (error) {
        _this2.setState({
          error: error
        });
      });
    }
  }, {
    key: "displayMainSlide",
    value: function displayMainSlide(_ref) {
      var mobile = _ref.mobile;
      var scoreData = this.state.score.creditReportInfo;
      var slides = [{
        type: 'score',
        score: scoreData.score,
        maxScore: scoreData.maxScoreValue,
        color: '#a7ced1'
      }, {
        type: 'debt',
        score: scoreData.currentLongTermDebt,
        maxScore: scoreData.currentLongTermDebt * 2,
        color: '#FCD29F',
        change: scoreData.changeInLongTermDebt
      }];
      var top;

      if (mobile) {
        top = 'calc(4vh)';
      } else {
        top = 'calc(50vh - 150px)';
      }

      return _react.default.createElement(_Slide.default, {
        animate: true,
        size: "big",
        slides: slides,
        style: {
          top: top,
          left: 'calc(50vw - 150px)'
        }
      });
    }
  }, {
    key: "displayOffersSlide",
    value: function displayOffersSlide() {
      var slides = [{
        type: 'offers',
        score: 5,
        maxScore: 5,
        color: '#fff'
      }];
      return _react.default.createElement(_Slide.default, {
        animate: false,
        size: "small",
        slides: slides,
        style: {
          top: 'calc(50vh - 250px)',
          left: 'calc(50vw - 320px)'
        }
      });
    }
  }, {
    key: "displayBalanceSlide",
    value: function displayBalanceSlide() {
      if (window.innerHeight < 520) {
        return null;
      }

      var slides = [{
        type: 'balance',
        score: 1,
        maxScore: 1,
        color: '#fff'
      }];
      return _react.default.createElement(_Slide.default, {
        animate: false,
        size: "small",
        slides: slides,
        style: {
          top: 'calc(90vh - 140px)',
          left: 'calc(50vw - 85px)'
        }
      });
    }
  }, {
    key: "displayMobileView",
    value: function displayMobileView() {
      return _react.default.createElement("div", null, this.displayMainSlide({
        mobile: true
      }), this.displayBalanceSlide());
    }
  }, {
    key: "displayNonMobileView",
    value: function displayNonMobileView() {
      return _react.default.createElement("div", null, this.displayMainSlide({
        mobile: false
      }), this.displayOffersSlide());
    }
  }, {
    key: "render",
    value: function render() {
      if (this.state.error) {
        return 'There has been an error. Please try again later.';
      } else if (!this.state.score) {
        return null;
      }

      var content;

      if (window.innerWidth <= _Constants.breakpoints.TABLET) {
        content = this.displayMobileView();
      } else {
        content = this.displayNonMobileView();
      }

      return _react.default.createElement("div", {
        className: cn(null, 'main')
      }, content);
    }
  }]);

  _inherits(Dashboard, _Component);

  return Dashboard;
}(_react.Component);

exports.default = Dashboard;
module.exports = exports["default"];