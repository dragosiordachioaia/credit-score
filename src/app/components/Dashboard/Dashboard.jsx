import React, { Component } from 'react';
import bemHelper from '../../utils/bem';

import { getJSON } from '../../utils/fetch';

import { SCORE_DATA_URL, breakpoints } from '../../constants/Constants';
import Slide from '../Slide/Slide';
import './dashboard.scss';

const cn = bemHelper({ block: 'dashboard' });

export default class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      score: null,
      error: null,
      windowWidth: 0,
    };

    this.fetchScoreData = this.fetchScoreData.bind(this);
    this.displayMainSlide = this.displayMainSlide.bind(this);
    this.displayMobileView = this.displayMobileView.bind(this);
    this.displayNonMobileView = this.displayNonMobileView.bind(this);
    this.onWindowResize = this.onWindowResize.bind(this);
  }

  componentDidMount() {
    if(!this.state.score && !this.state.error) {
      this.fetchScoreData();
    }
    if(window) {
      window.addEventListener("resize", this.onWindowResize);
    }
  }

  onWindowResize() {
    this.setState({windowWidth: window.innerWidth});
  }

  fetchScoreData() {
    getJSON(SCORE_DATA_URL).then(
      response => {
        this.setState({score: response});
      },
      error => {
        this.setState({error});
      }
    ).catch(error => {
      this.setState({error});
    });
  }

  displayMainSlide({mobile}) {
    const scoreData = this.state.score.creditReportInfo;
    const slides = [
      {
        type: 'score',
        score: scoreData.score,
        maxScore: scoreData.maxScoreValue,
        color: '#a7ced1',
      },
      {
        type: 'debt',
        score: scoreData.currentLongTermDebt,
        maxScore: scoreData.currentLongTermDebt * 2,
        color: '#FCD29F',
        change: scoreData.changeInLongTermDebt
      }
    ];

    let top;
    if(mobile) {
      top = 'calc(4vh)';
    } else {
      top = 'calc(50vh - 150px)';
    }

    return (
      <Slide
        animate={true}
        size='big'
        slides={slides}
        style={{
          top,
          left: 'calc(50vw - 150px)'
        }}
      />
    );
  }

  displayOffersSlide() {
    const slides = [{
      type: 'offers',
      score: 5,
      maxScore: 5,
      color: '#fff',
    }];
    return (
      <Slide
        animate={false}
        size='small'
        slides={slides}
        style={{
          top: 'calc(50vh - 250px)',
          left: 'calc(50vw - 320px)'
        }}
      />
    );
  }

  displayBalanceSlide() {
    if(window.innerHeight < 520) {
      return null;
    }

    const slides = [{
      type: 'balance',
      score: 1,
      maxScore: 1,
      color: '#fff',
    }]
    return (
      <Slide
        animate={false}
        size='small'
        slides={slides}
        style={{
          top: 'calc(90vh - 140px)',
          left: 'calc(50vw - 85px)'
        }}
      />
    );
  }

  displayMobileView() {
    return (
      <div>
        {this.displayMainSlide({mobile: true})}
        {this.displayBalanceSlide()}
      </div>
    );
  }

  displayNonMobileView() {
    return (
      <div>
        {this.displayMainSlide({mobile: false})}
        {this.displayOffersSlide()}
      </div>
    );
  }

  render() {
    if(this.state.error) {
      return 'There has been an error. Please try again later.';
    } else if(!this.state.score) {
      return null;
    }

    let content;
    if(window.innerWidth <= breakpoints.TABLET) {
      content = this.displayMobileView();
    } else {
      content = this.displayNonMobileView();
    }

    return (
      <div className={cn(null, 'main')}>
        {content}
      </div>
    )
  }
}
