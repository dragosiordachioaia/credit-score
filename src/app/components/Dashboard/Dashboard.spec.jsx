import React from 'react';
import renderer from 'react-test-renderer';

import Dashboard from './Dashboard';

let component;

const scoreData = {
  creditReportInfo: {
    score: 514,
    maxScoreValue: 700,
    currentLongTermDebt: 1000,
    changeInLongTermDebt: -100,
  }
};

describe('Dashboard component', () => {
  it('fetches data when loaded', () => {
    const component = renderer.create(
      <Dashboard />
    );
    const instance = component.getInstance();
    instance.fetchScoreData = jest.fn();
    instance.componentDidMount();
    expect(instance.fetchScoreData).toBeCalled();
  });
  it('onWindowResize changes state', () => {
    const component = renderer.create(
      <Dashboard />
    );
    const instance = component.getInstance();
    instance.setState = jest.fn();
    instance.onWindowResize();
    expect(instance.setState).toBeCalled();
  });
  it('displayMainSlide renders correctly on mobile', () => {
    const component = renderer.create(
      <Dashboard />
    );
    const instance = component.getInstance();
    instance.state.score = scoreData;
    const result = instance.displayMainSlide({mobile: true});
    expect(result).toMatchSnapshot();
  });
  it('displayMainSlide renders correctly on non mobile', () => {
    const component = renderer.create(
      <Dashboard />
    );
    const instance = component.getInstance();
    instance.state.score = scoreData;
    const result = instance.displayMainSlide({mobile: false});
    expect(result).toMatchSnapshot();
  });
  it('displayOffersSlide renders correctly', () => {
    const component = renderer.create(
      <Dashboard />
    );
    const instance = component.getInstance();
    instance.state.score = scoreData;
    const result = instance.displayOffersSlide();
    expect(result).toMatchSnapshot();
  });
  it('displayBalanceSlide renders correctly when height is small', () => {
    const component = renderer.create(
      <Dashboard />
    );
    global.innerHeight = 320;
    const instance = component.getInstance();
    instance.state.score = scoreData;
    const result = instance.displayBalanceSlide();
    expect(result).toMatchSnapshot();
  });
  it('displayBalanceSlide renders correctly when height is big enough', () => {
    const component = renderer.create(
      <Dashboard />
    );
    global.innerHeight = 800;
    const instance = component.getInstance();
    instance.state.score = scoreData;
    const result = instance.displayBalanceSlide();
    expect(result).toMatchSnapshot();
  });
  it('correctly renders error', () => {
    const component = renderer.create(
      <Dashboard />
    );

    const instance = component.getInstance();
    instance.state.error = true;
    const result = instance.render();
    expect(result).toMatchSnapshot();
  });
  it('correctly renders nothing when no score and no error', () => {
    const component = renderer.create(
      <Dashboard />
    );

    const instance = component.getInstance();
    instance.state.error = null;
    instance.state.score = null;
    const result = instance.render();
    expect(result).toMatchSnapshot();
  });
  it('correctly displays mobile view', () => {
    const component = renderer.create(
      <Dashboard />
    );

    const instance = component.getInstance();
    global.innerWidth = 320;
    instance.state.score = {};
    instance.displayMobileView = jest.fn().mockReturnValue('mobile_view');
    instance.displayNonMobileView = jest.fn().mockReturnValue('non_mobile_view');
    const result = instance.render();
    expect(result).toMatchSnapshot();
  });

  it('correctly displays non-mobile view', () => {
    const component = renderer.create(
      <Dashboard />
    );

    const instance = component.getInstance();
    global.innerWidth = 1024;
    instance.state.score = {};
    instance.displayMobileView = jest.fn().mockReturnValue('mobile_view');
    instance.displayNonMobileView = jest.fn().mockReturnValue('non_mobile_view');
    const result = instance.render();
    expect(result).toMatchSnapshot();
  });

  it('mobile view displays the right contents', () => {
    const component = renderer.create(
      <Dashboard />
    );

    const instance = component.getInstance();
    instance.displayMainSlide = jest.fn().mockReturnValue('main_slide');
    instance.displayBalanceSlide = jest.fn().mockReturnValue('balance_slide');
    const result = instance.displayMobileView();
    expect(result).toMatchSnapshot();
  });

  it('non-mobile view displays the right contents', () => {
    const component = renderer.create(
      <Dashboard />
    );

    const instance = component.getInstance();
    instance.displayMainSlide = jest.fn().mockReturnValue('main_slide');
    instance.displayOffersSlide = jest.fn().mockReturnValue('offers_slide');
    const result = instance.displayNonMobileView();
    expect(result).toMatchSnapshot();
  });
});
