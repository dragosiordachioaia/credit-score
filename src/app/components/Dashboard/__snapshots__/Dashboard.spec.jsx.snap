// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component correctly displays mobile view 1`] = `
<div
  className="dashboard dashboard--main"
>
  mobile_view
</div>
`;

exports[`Dashboard component correctly displays non-mobile view 1`] = `
<div
  className="dashboard dashboard--main"
>
  non_mobile_view
</div>
`;

exports[`Dashboard component correctly renders error 1`] = `"There has been an error. Please try again later."`;

exports[`Dashboard component correctly renders nothing when no score and no error 1`] = `null`;

exports[`Dashboard component displayBalanceSlide renders correctly when height is big enough 1`] = `
<Slide
  animate={false}
  size="small"
  slides={
    Array [
      Object {
        "color": "#fff",
        "maxScore": 1,
        "score": 1,
        "type": "balance",
      },
    ]
  }
  style={
    Object {
      "left": "calc(50vw - 85px)",
      "top": "calc(50vh + 80px)",
    }
  }
/>
`;

exports[`Dashboard component displayBalanceSlide renders correctly when height is small 1`] = `null`;

exports[`Dashboard component displayMainSlide renders correctly on mobile 1`] = `
<Slide
  animate={true}
  size="big"
  slides={
    Array [
      Object {
        "color": "#a7ced1",
        "maxScore": 700,
        "score": 514,
        "type": "score",
      },
      Object {
        "change": -100,
        "color": "#FCD29F",
        "limit": 0,
        "maxScore": 9007199254740991,
        "score": 1000,
        "type": "debt",
      },
    ]
  }
  style={
    Object {
      "left": "calc(50vw - 150px)",
      "top": "calc(50vh - 250px)",
    }
  }
/>
`;

exports[`Dashboard component displayMainSlide renders correctly on non mobile 1`] = `
<Slide
  animate={true}
  size="big"
  slides={
    Array [
      Object {
        "color": "#a7ced1",
        "maxScore": 700,
        "score": 514,
        "type": "score",
      },
      Object {
        "change": -100,
        "color": "#FCD29F",
        "limit": 0,
        "maxScore": 9007199254740991,
        "score": 1000,
        "type": "debt",
      },
    ]
  }
  style={
    Object {
      "left": "calc(50vw - 150px)",
      "top": "calc(50vh - 150px)",
    }
  }
/>
`;

exports[`Dashboard component displayOffersSlide renders correctly 1`] = `
<Slide
  animate={false}
  size="small"
  slides={
    Array [
      Object {
        "color": "#fff",
        "maxScore": 5,
        "score": 5,
        "type": "offers",
      },
    ]
  }
  style={
    Object {
      "left": "calc(50vw - 320px)",
      "top": "calc(50vh - 250px)",
    }
  }
/>
`;

exports[`Dashboard component mobile view displays the right contents 1`] = `
<div>
  main_slide
  balance_slide
</div>
`;

exports[`Dashboard component non-mobile view displays the right contents 1`] = `
<div>
  main_slide
  offers_slide
</div>
`;
